### DISCUSSION TITLE
Seting descriptor modbus master

### DISCUSSION THREAD

**Discussion by Silvesterrr** [espressif/esp-modbus#68] [General]
As I assume setting descriptor in modbus master rtu instance should be done once?
I've realized that was the idea of library author is that correct?

Does using function ```mbc_master_set_descriptor``` every often in code execution does come with any drawbacks? Like possible memory leaks?
From my quick assement it shouldn't cause any trouble.

You can correct me if I'm wrong.

using 1.0.15 esp-modbus

**Comment by alisitsyn** [comment]
@Silvesterrr, 

> As I assume setting descriptor in modbus master rtu instance should be done once?
I've realized that was the idea of library author is that correct?

This design idea came from the start and was defined as a requirement. The typical use case for Modbus master is: 1. setup the objects and network then 2. start communication. For the esp-modbus v1.0.x serial the `mbc_master_set_descriptor` is relatively simple and can update the object dictionary run time but it may cause issues. Note that the modbus task has access to data dictionary means if the data dictionary is updated when the stack is active it may cause serious problems such as memory leaks and even crash. In this case the access to the data dictionary should be protected by mutex to use it safely by synchronization task and modbus task.

Note: the esp-modbus v2.0 tcp stack uses smarter logic to check the object dictionary with conjunction with address table. It can not be changed simply run time.
